### ISSUE TITLE
Fix AdamW epsilon placement

### ISSUE THREAD

**Issue by MarcosAsh** [MarcosAsh/3dFluidDynamicsInC#123]
In \`optimizer.cpp:76\`, the AdamW update computes:

\`\`\`cpp
double update = m_hat / (std::sqrt(v_hat) + eps);
\`\`\`

The standard formulation adds epsilon inside the square root:

\`\`\`cpp
double update = m_hat / std::sqrt(v_hat + eps);
\`\`\`

The current form can produce large updates when \`v_hat\` is very small, because \`sqrt(v_hat)\` approaches zero and the denominator becomes just \`eps\` (1e-8). The correct form ensures the denominator never drops below \`sqrt(eps)\` (~1e-4), which is a much safer floor.

This matches the original Adam paper (Kingma & Ba, 2014) and what PyTorch/JAX implement.

The model trains fine currently because the learning rate and gradient clipping compensate, but this should be fixed before tuning hyperparameters or training larger models.
